tests/tests_pytorch/helpers/datamodules.py

```diff
@@ -20,7 +20,7 @@
 from pytorch_lightning.core.datamodule import LightningDataModule
 from tests_pytorch.helpers.datasets import MNIST, SklearnDataset, TrialMNIST
 
-_SKLEARN_AVAILABLE = RequirementCache("sklearn")
+_SKLEARN_AVAILABLE = RequirementCache("scikit-learn")
```

Comment:
This is reverting #14475 which was reverting already the other way. 
Either we agree to check for the module, or we agree to check for the package. These two things can have different names. RequirementCache is for the package, module_available is for the module. Correct @carmocca ?